client/components/onboarding-select-control/index.js

```diff
@@ -0,0 +1,195 @@
+	{ selectedItem },
+	{ type, changes, props: { items } }
+) => {
+	switch ( type ) {
+		case useSelect.stateChangeTypes.ToggleButtonKeyDownArrowDown:
+			// If we already have a selected item, try to select the next one,
+			// without circular navigation. Otherwise, select the first item.
+			return {
+				selectedItem:
+					items[
+						selectedItem
+							? Math.min(
+									items.indexOf( selectedItem ) + 1,
+									items.length - 1
+							  )
+							: 0
+					],
+			};
+		case useSelect.stateChangeTypes.ToggleButtonKeyDownArrowUp:
+			// If we already have a selected item, try to select the previous one,
+			// without circular navigation. Otherwise, select the last item.
+			return {
+				selectedItem:
+					items[
+						selectedItem
+							? Math.max( items.indexOf( selectedItem ) - 1, 0 )
+							: items.length - 1
+					],
+			};
+		default:
+			return changes;
+	}
+};
+export default function OnboardingSelectControl( {
+	className,
+	label,
+	// describedBy,
+	options: items,
+	onChange: onSelectedItemChange,
+	value: _selectedItem,
+} ) {
+	const {
+		getLabelProps,
+		getToggleButtonProps,
+		getMenuProps,
+		getItemProps,
+		isOpen,
+		highlightedIndex,
+		selectedItem,
+	} = useSelect( {
+		initialSelectedItem: items[ 0 ],
+		items,
+		itemToString,
+		onSelectedItemChange,
+		...( 'undefined' !== typeof _selectedItem && null !== _selectedItem
+			? { selectedItem: _selectedItem }
+			: undefined ),
+		stateReducer,
+	} );
+
+	// function getDescribedBy() {
+	// 	if ( describedBy ) {
+	// 		return describedBy;
+	// 	}
+
+	// 	if ( ! selectedItem ) {
+	// 		return __( 'No selection' );
+	// 	}
+
+	// 	// translators: %s: The selected option.
+	// 	return sprintf( __( 'Currently selected: %s' ), selectedItem.name );
+	// }
+
+	const menuProps = getMenuProps( {
+		className: 'components-custom-select-control__menu',
+		'aria-hidden': ! isOpen,
+	} );
+
+	const onKeyDownHandler = useCallback(
+		( e ) => {
+			e.stopPropagation();
+			menuProps?.onKeyDown?.( e );
+		},
+		[ menuProps ]
+	);
+
+	// We need this here, because the null active descendant is not fully ARIA compliant.
+	if (
+		menuProps[ 'aria-activedescendant' ]?.startsWith( 'downshift-null' )
+	) {
+		delete menuProps[ 'aria-activedescendant' ];
+	}
+	return (
+		<div
+			className={ classNames(
+				'components-custom-select-control',
+				className
+			) }
+		>
+			{
+				/* eslint-disable-next-line jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for */
+				<label
+					{ ...getLabelProps( {
+						className: 'components-custom-select-control__label',
+					} ) }
+				>
+					{ label }
+				</label>
+			}
+			<Button
+				{ ...getToggleButtonProps( {
+					// This is needed because some speech recognition software don't support `aria-labelledby`.
+					'aria-label': label,
+					'aria-labelledby': undefined,
+					className: 'components-custom-select-control__button',
+					isSmall: true,
+					// describedBy: getDescribedBy(),
```

Comment:
I needed to comment out this line to prevent a console error - not sure why, as this was directly copied from the [CustomSelectControl](https://github.com/WordPress/gutenberg/tree/7aa042605ff42bb437e650c39132c0aa8eb4ef95/packages/components/src/custom-select-control).

Reply:
That prop should be `aria-describedby`, maybe our version of `@wordpress/components` doesn't have `describedBy` prop defined in `<Button/>`.

Reply:
Thanks! Addressed in c7dd051342c8a1e6fde44fa9ebc488e41479ce30.